src/Midgard/CreatePHP/Mapper/DoctrineOrmMapper.php

```diff
@@ -37,6 +37,36 @@ class DoctrineOrmMapper extends BaseDoctrineRdfMapper
     /**
      * {@inheritDoc}
      *
+     * The ORM implementation for preparing the object
+     */
+    public function prepareObject(TypeInterface $type, $parent = null)
+    {
+        $object = parent::prepareObject($type);
+
+        if (null == $parent){
+            throw new RuntimeException('You need a parent to create new objects');
+        }
+
+        /** @var \Doctrine\ORM\Mapping\ClassMetadata $metaData */
+        $metaData = $this->om->getClassMetaData(get_class($object));
+        $metaDataParent = $this->om->getClassMetaData(get_class($parent));
+
+        foreach ($metaData->associationMappings as $mapping)
```

Comment:
how would you parametrize `findParentMapping`? i was thinking about to use `$metaData` as a parameter but that might be a bad idea if someone wants to override this method although i would have to retrieve `$metaData`twice. Should I use `$object` or just the string from `ClassUtils:getRealClass`? hard for me to imagine which cases might occur that require a customized version of `findParentMapping`...